### PR TITLE
compile_clang: Get the clang version from current AOSP documentation

### PR DESCRIPTION
**Pull request by pablomh**
Rely on the information provided in the prebuild clang README.

### CODE REVIEW

File: compile_clang.sh [hunk 1]
```diff
@@ -2,8 +2,9 @@
 
 _cross_compile=$(realpath "$ANDROID_ROOT/prebuilts/gcc/linux-x86/aarch64/aarch64-linux-android-4.9/bin/aarch64-linux-android-")
 _cross_compile_32=$(realpath "$ANDROID_ROOT/prebuilts/gcc/linux-x86/arm/arm-linux-androideabi-4.9/bin/arm-linux-androideabi-")
-_clang_path=$(realpath "$ANDROID_ROOT/prebuilts/clang/host/linux-x86/")
-_clang_path=$(ls -1d "$_clang_path/clang-r"*/bin | tail -1)
+# Get the clang version from current AOSP documentation
+_clang_version=$(grep -A1 'grep -A1 '^\* \[\*\*Android Linux Kernel' $ANDROID_ROOT/prebuilts/clang/host/linux-x86/README.md | awk '/^  \* Currently clang-/ {print $NF}')
```
Comment (MarijnS95) [inline compile_clang.sh]: I think you intended this to be:

```suggestion
_clang_version=$(grep -A1 '^\* \[\*\*Android Linux Kernel' $ANDROID_ROOT/prebuilts/clang/host/linux-x86/README.md | awk '/^  \* Currently clang-/ {print $NF}')
```
?

Otherwise:
```
/data/aosp-r/oot/compile_clang.sh: line 6: unexpected EOF while looking for matching `''
```
And obviously grepping a stringified comment doesn't make a whole lot of sense in this context :)

**Comment by pablomh** [comment]
Trying this ugly GH feature. If you find it as ugly as ugly as I do I'll force-push when I have the chance later:)

**Comment by MarijnS95** [comment]
The feature is ugly, and I only use it to ge an automatic inline diff. GitHub should really have the option to find the unique commit the comment range applies to, and (if found) offer to amend it there instead. I can squash and push this for you too.

Meanwhile I'd like the match to be done with a single command. The regex for this should be _really_ simple: `Linux Kernel.*\n  \* Currently (clang-\S+)$` and only capturing group `\1` is printed. Unfortunately none of the tools (grep, awk, sed) seem to properly support multiline matching _and_ printing of capture groups (without writing many lines of inline commands). Do you happen to know how to do this easily?

**Comment by pablomh** [comment]
In that case I'll force-push it clean.

I've been investigating a little and the easier way I've found the match can be done is this:

`awk '/^\* \[\*\*Android Linux Kernel/,/^  \* Currently clang/ $0 ~ /Android Linux Kernel/ skip {if ($0 ~ /Currently/) {print $NF}}' $ANDROID_ROOT/prebuilts/clang/host/linux-x86/README.md`

**Comment by MarijnS95** [comment]
That is - unfortunately - the kind of magic command mode I want to prevent, including duplicating parts of the match :/

**Comment by pablomh** [comment]
Slightly better but still containing duplicated code:

`awk '/^\* \[\*\*Android Linux Kernel/,/^  \* Currently clang/ {if ($0 ~ /clang-/) {print $NF}}'`

**Comment by MarijnS95** [comment]
That's getting closer. I played around with the same pattern range match, but there seems to be no way to turn `(clang-\S+)` in the end pattern into a group and print it that way.

**Comment by pablomh** [comment]
Current state of the art: `$ awk 'f{print $NF} /^\* \[\*\*Android Linux Kernel/{f=1} /^  \* Currently clang-/{f=0}' prebuilts/clang/host/linux-x86/README.md`

**Comment by MarijnS95** [comment]
That's an interesting one, but still a bit too much magic for me. Basically it prints the last column (`$NF`) until finding `Currently clang-`, which'll mess up if new lines are added above. If we really intend to print the next line after `Android Linux Kernel` we should just print and reset it immediately:

```awk
f{print $NF; f=0} /^\* \[\*\*Android Linux Kernel/{f=1}
```

But then there's still a mess when `Currently clang-` is not immediately on the next line. But we have an easy solution to that... Presenting:

```awk
/^\* \[\*\*Android Linux Kernel/{f=1} /^  \* Currently clang-/ && f {print $NF; f=0}
```

Basically this only prints NF if an earlier line matches `Android Linux Kernel` and the current line contains the version!

To be honest I'd rather fail if there's more lines between `Android Linux Kernel` and `Currently clang-`, assuming the format or indentation might have inadvertently been changed and we'd end up selecting a `^  \* Currently clang-` from another block down below. For that, we can use `NR`. As a safety and optimization, we can call `exit` after printing as well:

```awk
/^\* \[\*\*Android Linux Kernel/{f=NR} /^  \* Currently clang-/ && f==NR-1 {print $NF; exit}
```

**Review by MarijnS95 — APPROVED** [review]
Great @pablomh, thanks! Still feel defeated we didn't get the simple multiline regex with capturing group working (instead having to literally program with counter variables)... but this works for now!
